core/trino-main/src/main/java/io/trino/operator/WorkProcessorOperatorAdapter.java

```diff
@@ -185,15 +185,20 @@ public void addInput(Page page)
     @Override
     public Page getOutput()
     {
-        if (!pages.process()) {
-            return null;
-        }
+        try {
+            if (!pages.process()) {
+                return null;
+            }
 
-        if (pages.isFinished()) {
-            return null;
-        }
+            if (pages.isFinished()) {
+                return null;
+            }
 
-        return pages.getResult();
+            return pages.getResult();
+        }
+        finally {
+            operatorContext.setLatestMetrics(workProcessorOperator.getMetrics());
```

Comment:
if this throws, this may swallow an exception from the `try` block.
do we have to use `try finally` here?

Reply:
removed `try finally`